google/cloud/storage/internal/generate_message_boundary.h

```diff
@@ -58,6 +67,24 @@ template <typename RandomStringGenerator,
 std::string GenerateMessageBoundary(
     std::string const& message, RandomStringGenerator&& random_string_generator,
     int initial_size, int growth_size) {
+  // This is not the only place we make this assumption. Duplicating the static
+  // assertion should make it easier to find the code if we ever need to fix it.
+  static_assert(std::numeric_limits<unsigned char>::max() == 255,
+                "required for efficient GenerateMessageBoundary");
+
+  static char const kBoundaryChars[] =
+      "abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  auto constexpr kSkip = std::size_t{64};
+  std::array<bool, 255> seen;
+  std::fill(seen.begin(), seen.end(), false);
+  for (auto i = kSkip - 1; i < message.size(); i += kSkip) {
+    seen[static_cast<unsigned char>(message[i])] = true;
+  }
+  for (auto c : kBoundaryChars) {
+    if (c == '\0') continue;
```

Comment:
Nit: I don't have a nice workaround for this, but it seems like `kBoundaryChars` is not living up to its name when this check is required.